Kernel/Locking/Mutex.h

```diff
@@ -93,12 +94,16 @@ class Mutex {
     // When locked exclusively, this is always the one thread that holds the
     // lock.
     RefPtr<Thread> m_holder;
-    HashMap<Thread*, u32> m_shared_holders;
+    size_t m_shared_holders { 0 };
```

Comment:
Is this a race? I am thinking this should be `AK::Atomic<size_t>` so that the `fetch_add` happens safely. This would also mean that the result of the increment/decrement would need to be checked. Maybe I am missing something?

Reply:
Mutex functions are locked using a spinlock